src/abstractsparsearray.jl

```diff
@@ -53,3 +53,63 @@ function getunstoredindex(a::ReplacedUnstoredSparseArray, I::Int...)
 end
 eachstoredindex(a::ReplacedUnstoredSparseArray) = eachstoredindex(parent(a))
 @derive ReplacedUnstoredSparseArray AbstractArrayOps
+
+# Special-purpose constructors
+# ----------------------------
+using Random: Random, AbstractRNG, default_rng
+
+@doc """
+    spzeros([T::Type], dims) -> A::SparseArrayDOK{T}
+
+Create an empty size `dims` sparse array.
+The optional `T` argument specifies the element type, which defaults to `Float64`.
+""" spzeros
+
+spzeros(dims::Dims) = spzeros(Float64, dims)
+spzeros(::Type{T}, dims::Dims) where {T} = SparseArrayDOK{T}(undef, dims)
+
+@doc """
+    sprand([rng], [T::Type], dims; density::Real=0.5, rfn::Function=rand) -> A::SparseArrayDOK{T}
+
+Create a random size `dims` sparse array in which the probability of any element being stored is independently given by `density`.
+The optional `rng` argument specifies a random number generator, see also `Random`.
+The optional `T` argument specifies the element type, which defaults to `Float64`.
+The optional `rfn` argument can be used to control the type of random elements.
+
+See also [`sprand!`](@ref).
+""" sprand
+
+function sprand(::Type{T}, dims::Dims; kwargs...) where {T}
+  return sprand(default_rng(), T, dims; kwargs...)
+end
+sprand(dims::Dims; kwargs...) = sprand(default_rng(), Float64, dims; kwargs...)
+function sprand(rng::AbstractRNG, dims::Dims; kwargs...)
+  return sprand(rng, Float64, dims; kwargs...)
+end
+function sprand(rng::AbstractRNG, ::Type{T}, dims::Dims; kwargs...) where {T}
+  A = SparseArrayDOK{T}(undef, dims)
+  sprand!(rng, A; kwargs...)
+  return A
+end
+
+@doc """
+    sprand!([rng], A::AbstractArray; density::Real=0.5, rfn::Function=rand) -> A
+
+Overwrite part of an array with random entries, where the probability of overwriting is independently given by `density`.
+The optional `rng` argument specifies a random number generator, see also `Random`.
+The optional `rfn` argument can be used to control the type of random elements.
+
+See also [`sprand`](@ref).
+""" sprand!
+
+sprand!(A::AbstractArray; kwargs...) = sprand!(default_rng(), A; kwargs...)
+function sprand!(
+  rng::AbstractRNG, A::AbstractArray; density::Real=0.5, rfn::Function=Random.rand
```

Comment:
I would prefer a longer name than `rfn` but can't think of anything I particularly like (I think even `rand_function` would be better).

Reply:
This is copied from `SparseArrays.jl`, hence the short name (I would also typically go for something longer). Here are the things I can come up with: `randf`, `rand_function`, `rand_generator`, `randomizer`. I'll change it to `rand_function`, but feel free to let me know if you like any of the others better.

Reply:
> This is copied from `SparseArrays.jl`, hence the short name (I would also typically go for something longer).

Yes I noticed that, though I think there it is an argument, not a keyword argument, so that name isn't part of the interface (if they had made it a keyword argument, I would have biased towards using the same name as them).

> Here are the things I can come up with: `randf`, `rand_function`, `rand_generator`, `randomizer`. I'll change it to `rand_function`, but feel free to let me know if you like any of the others better.

Other names I can think of are `randfun` or `randfunc`.

Reply:
Oh you are right, I didn't even notice that. I'm pretty much okay with everything I think.

Reply:
This is pretty arbitrary, but I see `randfun` used in the source code of Base for a similar purpose: https://github.com/JuliaLang/julia/blob/v1.11.3/stdlib/Random/src/normal.jl#L219 and I think it looks ok so let's go with that one.